File: Java/hellointerview/03-intervals/nonOverlappingIntervals.java
import java.util;

class Solution {
  public int nonOverlappingIntervals(int[][] intervals) {
    if (intervals.length == 0) {
      return 0;
    }

    Arrays.sort(intervals, (a, b) -> a[1] - b[1]);

    int end = intervals[0][1];
    int count = 1;

    for (int i = 1; i < intervals.length; i++) {
      if (intervals[i][0] >= end) {
        end = intervals[i][1];
        count++;
      }
    }

    return intervals.length - count;
  }
}

File: TS/hellointerview/intervals/nonOverlappingIntervals.ts
function nonOverlappingIntervals(intervals: number[][]): number {
  if (intervals.length === 0) {
    return 0;
  }
  intervals.sort((a, b) => a[1] - b[1]);

  let end = intervals[0][1];
  let count = 1;

  for (let i = 1; i < intervals.length; i++) {
    if (intervals[i][1] >= end) {
      count++;
      end = intervals[i][1];
    }
  }

  return count;
}
